src/modules/joke.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * Modules:
+ *  - {@link joke}
+ * Description:
+ *  Uses the joke api provided by arc to parse a response and send it as a discord embed
+ */
+
+import * as util from '../core/util.js';
+import {botConfig} from '../core/config.js';
+
+// Config manager for nsfw detection
+const jokeConfig = botConfig.modules.joke;
+let JOKE_API_URL = '';
+if (jokeConfig.nsfw) {
+  JOKE_API_URL =
+    'https://v2.jokeapi.dev/joke/Any?blacklistFlags=religious,political,racist,sexist,explicit&type=single';
+} else {
+  JOKE_API_URL =
+    'https://v2.jokeapi.dev/joke/Any?blacklistFlags=nsfw,religious,political,racist,sexist,explicit&type=single';
+}
+
+const joke = new util.RootModule(
+  'joke', // command name
+  'Get a funny joke from the bot', // command description
+  [], // dependencies
+  [], // options
+
+  async (args, interaction) => {
+    fetchJoke()
+      .then(joke => {
```

Comment:
Please prefer `await` to `.then()` here.
```typescript
const joke = await fetchJoke();
```
It reduces the amount of nesting, and because `await` is locally blocking, it helps keep control flow clear.